src/frontend/src/optimizer/plan_node/generic/log_scan.rs

```diff
@@ -0,0 +1,230 @@
+    pub fn output_column_ids(&self) -> Vec<ColumnId> {
+        self.output_col_idx
+            .iter()
+            .map(|i| self.get_table_columns()[*i].column_id)
+            .collect()
+    }
+
+    pub fn output_column_ids_to_batch(&self) -> Vec<ColumnId> {
+        self.output_col_idx
+            .iter()
+            .map(|i| self.get_table_columns()[*i].column_id)
+            .filter(|i| i != &self.op_column.column_id)
+            .collect()
+    }
+
+    pub fn primary_key(&self) -> &[ColumnOrder] {
+        &self.table_desc.pk
+    }
+
+    pub(crate) fn column_names_with_table_prefix(&self) -> Vec<String> {
+        self.output_col_idx
+            .iter()
+            .map(|&i| format!("{}.{}", self.table_name, self.get_table_columns()[i].name))
+            .collect()
+    }
+
+    pub(crate) fn column_names(&self) -> Vec<String> {
+        println!("output_col_idx: {:?}", self.output_col_idx);
+        self.output_col_idx
+            .iter()
+            .map(|&i| self.get_table_columns()[i].name.clone())
+            .collect()
+    }
+
+    pub fn distribution_key(&self) -> Option<Vec<usize>> {
+        let tb_idx_to_op_idx = self
+            .output_col_idx
+            .iter()
+            .enumerate()
+            .map(|(op_idx, tb_idx)| (*tb_idx, op_idx))
+            .collect::<HashMap<_, _>>();
+        self.table_desc
+            .distribution_key
+            .iter()
+            .map(|&tb_idx| tb_idx_to_op_idx.get(&tb_idx).cloned())
+            .collect()
+    }
+
+    /// get the Mapping of columnIndex from internal column index to output column index
+    pub fn i2o_col_mapping(&self) -> ColIndexMapping {
+        ColIndexMapping::with_remaining_columns(
+            &self.output_col_idx,
+            self.get_table_columns().len(),
+        )
+    }
+
+    /// Get the ids of the output columns and primary key columns.
+    pub fn output_and_pk_column_ids(&self) -> Vec<ColumnId> {
+        let mut ids = self.output_column_ids();
+        for column_order in self.primary_key() {
+            let id = self.get_table_columns()[column_order.column_index].column_id;
+            if !ids.contains(&id) {
+                ids.push(id);
+            }
+        }
+        ids
+    }
+
+    /// Create a logical scan node for log table scan
+    pub(crate) fn new(
+        table_name: String,
+        output_col_idx: Vec<usize>,
+        table_desc: Rc<TableDesc>,
+        op_column: Rc<ColumnDesc>,
+        ctx: OptimizerContextRef,
+        old_epoch: u64,
+        new_epoch: u64,
+    ) -> Self {
+        Self {
+            table_name,
+            output_col_idx,
+            table_desc,
+            op_column,
+            chunk_size: None,
+            ctx,
+            old_epoch,
+            new_epoch,
+        }
+    }
+
+    pub(crate) fn columns_pretty<'a>(&self, verbose: bool) -> Pretty<'a> {
+        Pretty::Array(
+            match verbose {
+                true => self.column_names_with_table_prefix(),
```

Comment:
No need to have two implementations. Just keep one of them and use it anyway no matter whether verbose or not.

Reply:
It looks like verbose is a configuration common to all plannode, do we need to leave it alone here?

Reply:
It's up to you. Previously the comment is added just to reduce the usage of `get_table_columns`, but now we can use `self.table_desc.columns` easily, so both are fine.